### PR TITLE
connman: 1.41 -> 1.42

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Bump version, remove patches that are part of the release, and use SRI hashes

Also replaced maintainer with myself as it seems they are part of the retired contributors group

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lf-** [comment]
btw your commit message has the wrong versions, it's 1.41 and 1.42 :)

**Comment by bjornfor** [comment]
> Bump version, use finalAttrs idiom instead of rec, remove patches that are part of the release, and use SRI hashes
>
> Also replaced maintainer with myself as it seems they are part of the retired contributors group

This sounds like the 1 commit should be split into 3:
* bump
* use finalAttrs pattern
* update maintainers

**Comment by eclairevoyant** [comment]
@lf- Thanks for catching, fixed.
@bjornfor After discussing elsewhere further, I removed the `finalAttrs` change as anyone overriding would need to provide the fetcher for `src` anyway (due to changed hash), therefore `finalAttrs` provides no benefit here in terms of overriding. Remaining changes have been split as requested.

**Comment by eclairevoyant** [comment]
@lf- @bjornfor are there any other blockers to merging this?

**Comment by bjornfor** [comment]
I don't have any objections, but I don't feel particularly qualified to approve/merge this either.

> connman: update maintainers 

Can you write in the commit message why you're removing the existing maintainer?

Does the patch not have an upstream we can fetch from? Will it be upstreamed? These are things worth writing a comment about in the code or commit message.

**Comment by lf-** [comment]
> Can you write in the commit message why you're removing the existing maintainer?
To my understanding it was because the previous maintainer is not working on the package for several versions.

> Does the patch not have an upstream we can fetch from? Will it be upstreamed? These are things worth writing a comment about in the code or commit message.

Upstream hecked up their dist tarball for 1.42. The fix was committed a few days ago so it will be fixed in the next release, we can just delete it after. The file is simply taken from the repo: #250307 is another approach to the same thing, by not using a dist tarball, instead just using the git tag.

**Comment by eclairevoyant** [comment]
@bjornfor I basically reused part of one of the previous patches because of the broken dist tarball, added comment for it. maintainer-related commit has also been updated as requested.

File: pkgs/tools/networking/connman/connman/default.nix [hunk 1]
```diff
@@ -59,28 +59,21 @@ let inherit (lib) optionals; in
...
-    })
-    (fetchpatch {
-      name = "CVE-2023-28488.patch";
-      url = "https://git.kernel.org/pub/scm/network/connman/connman.git/patch/?id=99e2c16ea1cced34a5dc450d76287a1c3e762138";
-      sha256 = "sha256-377CmsECji2w/c4bZXR+TxzTB7Lce0yo7KdK1oWfCVY=";
-    })
-  ] ++ lib.optionals stdenv.hostPlatform.isMusl [
+    # simply the middle section of upstream commit a48864a2e5d2a725dfc6eef567108bc13b43857f
+    # dist tarball is broken, hence this patch as a workaround
+    ./create-libppp-compat.h.patch
```
Comment (wegank) [inline pkgs/tools/networking/connman/connman/default.nix:71]: Is it possible to replace it with fetchpatch?

Comment (eclairevoyant) [reply pkgs/tools/networking/connman/connman/default.nix:71]: Unfortunately it's only 1/3 of the diff associated with that commit, the other files already have the patch applied

Comment (alyssais) [reply pkgs/tools/networking/connman/connman/default.nix:71]: For next time, it's possible to tell fetchpatch to exclude/include specific files.

Comment (eclairevoyant) [reply pkgs/tools/networking/connman/connman/default.nix:71]: didn't realise that, thanks.

File: pkgs/tools/networking/connman/connman/default.nix [hunk 2]
```diff
@@ -59,28 +59,21 @@ let inherit (lib) optionals; in
...
+  ] ++ optionals stdenv.hostPlatform.isMusl [
     # Fix Musl build by avoiding a Glibc-only API.
     (fetchpatch {
       url = "https://git.alpinelinux.org/aports/plain/community/connman/libresolv.patch?id=e393ea84386878cbde3cccadd36a30396e357d1e";
-      sha256 = "1kg2nml7pdxc82h5hgsa3npvzdxy4d2jpz2f93pa97if868i8d43";
+      hash = "sha256-7Q1bp8rD/gGVYUqnIXqjr9vypR8jlC926p3KYWl9kLw=";
```
Comment (alyssais) [inline pkgs/tools/networking/connman/connman/default.nix:76]: This hash is just not correct:

```
% nix --extra-experimental-features nix-command hash to-sri --type sha256 1kg2nml7pdxc82h5hgsa3npvzdxy4d2jpz2f93pa97if868i8d43
sha256-gzQUkUEunqTuSE78K0Ujvre/rx1KP1igQKy3e2i14s0=
```

Fix in https://github.com/NixOS/nixpkgs/pull/255103.

**Pull request merged**